stackslib/src/net/api/getsortition.rs

```diff
@@ -0,0 +1,304 @@
+    pub miner_pk_hash160: Option<Hash160>,
+    /// If sortition occurred, this will be the consensus hash of the burn block corresponding
+    /// to the winning block commit's parent block ptr. In 3.x, this is the consensus hash of
+    /// the tenure that this new burn block's miner will be building off of.
+    #[serde(with = "prefix_opt_hex")]
+    pub stacks_parent_ch: Option<ConsensusHash>,
+    /// If sortition occurred, this will be the consensus hash of the most recent sortition before
+    ///  this one.
+    #[serde(with = "prefix_opt_hex")]
+    pub last_sortition_ch: Option<ConsensusHash>,
+    #[serde(with = "prefix_opt_hex")]
+    /// In Stacks 2.x, this is the winning block.
+    /// In Stacks 3.x, this is the first block of the parent tenure.
+    pub committed_block_hash: Option<BlockHeaderHash>,
+}
+
+impl TryFrom<(&str, &str)> for QuerySpecifier {
+    type Error = Error;
+
+    fn try_from(value: (&str, &str)) -> Result<Self, Self::Error> {
+        let hex_str = if value.1.starts_with("0x") {
+            &value.1[2..]
+        } else {
+            value.1
+        };
+        match value.0 {
+            "consensus" => Ok(Self::ConsensusHash(
+                ConsensusHash::from_hex(hex_str).map_err(|e| Error::DecodeError(e.to_string()))?,
+            )),
+            "burn" => Ok(Self::BurnchainHeaderHash(
+                BurnchainHeaderHash::from_hex(hex_str)
+                    .map_err(|e| Error::DecodeError(e.to_string()))?,
+            )),
+            "burn_height" => Ok(Self::BlockHeight(
+                value
+                    .1
+                    .parse::<u64>()
+                    .map_err(|e| Error::DecodeError(e.to_string()))?,
+            )),
+            other => Err(Error::DecodeError(format!("Unknown query param: {other}"))),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct GetSortitionHandler {
+    pub query: QuerySpecifier,
+}
+
+impl GetSortitionHandler {
+    pub fn new() -> Self {
+        Self {
+            query: QuerySpecifier::Latest,
+        }
+    }
+}
+/// Decode the HTTP request
+impl HttpRequest for GetSortitionHandler {
+    fn verb(&self) -> &'static str {
+        "GET"
+    }
+
+    fn path_regex(&self) -> Regex {
+        Regex::new(PATH_REGEX).unwrap()
+    }
+
+    /// Try to decode this request.
+    /// There's nothing to load here, so just make sure the request is well-formed.
+    fn try_parse_request(
+        &mut self,
+        preamble: &HttpRequestPreamble,
+        captures: &Captures,
+        query: Option<&str>,
+        _body: &[u8],
+    ) -> Result<HttpRequestContents, Error> {
+        if preamble.get_content_length() != 0 {
+            return Err(Error::DecodeError(
+                "Invalid Http request: expected 0-length body".to_string(),
+            ));
+        }
+
+        let req_contents = HttpRequestContents::new().query_string(query);
+        self.query = QuerySpecifier::Latest;
+        eprintln!("{captures:?}");
```

Comment:
Thanks for fixing this path up!  I think you might want to remove the `eprintln!` though :)

Reply:
Removed in 7b1ad1e